plugins/event_source/rabbitmq.py

```diff
@@ -0,0 +1,75 @@
+"""
+rabbitmq.py
+An ansible-rulebook event source plugin for receiving events via RabbitMQ
+
+author:
+    - Timothy Test (@ttestscripting)
+
+Arguments:
+    host       The RabbitMQ host
+    port       The port for RabbitMQ
+               Default 5672
+    v_host:    Virtual Host
+               Default '/'
+    username:  The RabbitMQ username
+               Default guest
+    password:  The RabbitMQ password
+               Default guest
+    queue:     queue to watch
+"""
+
+import asyncio
+import logging
+import aio_pika
+from typing import Any, Dict
+
+async def main(queue: asyncio.Queue, args: Dict[str, Any]):
+
+    logger = logging.getLogger()
+
+    rabbitmq_host = args.get("host")
+    rabbitmq_port = args.get("port", "5672")
+    rabbitmq_v_host = args.get("v_host", "/")
+    rabbitmq_username = args.get("username", "guest")
+    rabbitmq_password = args.get("password", "guest")
+    rabbitmq_queue = args.get("queue")
+    # TODO add options for SSL
+    connection = await aio_pika.connect_robust(
+        host=rabbitmq_host,
+        login=rabbitmq_username,
+        password=rabbitmq_password,
+        port=rabbitmq_port,
+        virtualhost=rabbitmq_v_host
+    )
+
+    async with connection:
+
+        # Creating channel
+        channel: aio_pika.abc.AbstractChannel = await connection.channel()
+
+        # Declaring queue
+        mq_queue: aio_pika.abc.AbstractQueue = await channel.declare_queue(rabbitmq_queue)
+
+        async with mq_queue.iterator() as queue_iter:
+            async for message in queue_iter:
+                async with message.process():
+                    logger.info(message.body.decode())
+                    data = dict(
```

Comment:
We want all plugins to generate events in similar format. Please take a look at kafka.py as an example. The message body itself must be a json str and is converted into a dict. No need to use meta to store the information about the message bus itself for every event.

Reply:
Converted the format as requested